charts/karpenter/values.yaml

```diff
@@ -51,7 +51,10 @@ podDisruptionBudget:
   maxUnavailable: 1
 # -- SecurityContext for the pod.
 podSecurityContext:
+  runAsNonRoot: true
```

Comment:
Can we do this? I think we explicitly avoided doing this since it would change the security context of sidecar containers that may not support the more locked-down configuration

Reply:
This sets the default which can then be overridden by the consumer. Karpenter would be a high value target and this makes it really clear when adding a sidecar that running as root impacts the whole pod and not just the container.

Reply:
We did this and then pulled it back because multiple people ran into issues with it -- I'm fine having a more open discussion on this in something like a working group but I'm against breaking users in this way after the number of issues that were opened when we first tried to make this change (see https://github.com/aws/karpenter-provider-aws/pull/4681)

Reply:
I'm pretty weary of this, too since this is a breaking change to the chart post-v1 and I believe that folks will miss this during the upgrade process

Reply:
@jonathan-innis let's take this to a separate discussion, I'll change this to false for the PR.

That said I am very interested why it was reversed in the first place? It significantly increases the attack surface for all users, but off the top of my head I can't think of many valid uses for root sidecars? Furthermore anyone using sidecars as root should have the K8s contextual knowledge to be proactively checking the pod security spec of the targets.